src/System.Net.Primitives/tests/FunctionalTests/IPEndPointParsing.cs

```diff
@@ -0,0 +1,305 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Net.Sockets;
+using Xunit;
+
+namespace System.Net.Primitives.Functional.Tests
+{
+    public class IPEndPointParsing
+    {
+        [Theory]
+        [MemberData(nameof(IPAddressParsing.ValidIpv4Addresses), MemberType = typeof(IPAddressParsing))]    // Just borrow the list from IPAddressParsing
+        public void Parse_ValidEndPoint_IPv4_Success(string address, string expectedAddress)
+        {
+            Parse_ValidEndPoint_Success(address, expectedAddress, true);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidIpv6AddressesNoPort))]  // We need our own list here to explicitly exclude port numbers and brackets without making the test overly complicated (and less valid)
+        public void Parse_ValidEndPoint_IPv6_Success(string address, string expectedAddress)
+        {
+            Parse_ValidEndPoint_Success(address, expectedAddress, false);
+        }
+
+        private void Parse_ValidEndPoint_Success(string address, string expectedAddress, bool isIPv4)
+        {
+            // We'll parse just the address alone followed by the address with various port numbers
+
+            expectedAddress = expectedAddress.ToLowerInvariant();   // This is done in the IP parse routines
+
+            // TryParse should return true
+            Assert.True(IPEndPoint.TryParse(address, out IPEndPoint result));
+            Assert.Equal(expectedAddress, result.Address.ToString());
+            Assert.Equal(0, result.Port);
+
+            // Parse should give us the same result
+            result = IPEndPoint.Parse(address);
+            Assert.Equal(expectedAddress, result.Address.ToString());
+            Assert.Equal(0, result.Port);
+
+            // Cover varying lengths of port number
+            int portNumber = 1;
+            for (int i = 0; i < 5; i++)
+            {
+                var addressAndPort = isIPv4 ? $"{address}:{i}" : $"[{address}]:{i}";
+
+                // TryParse should return true
+                Assert.True(IPEndPoint.TryParse(addressAndPort, out result));
+                Assert.Equal(expectedAddress, result.Address.ToString());
+                Assert.Equal(i, result.Port);
+
+                // Parse should give us the same result
+                result = IPEndPoint.Parse(addressAndPort);
+                Assert.Equal(expectedAddress, result.Address.ToString());
+                Assert.Equal(i, result.Port);
+
+                // i.e.: 1; 12; 123; 1234; 12345
+                portNumber *= 10;
+                portNumber += i + 2;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IPAddressParsing.InvalidIpv4Addresses), MemberType = typeof(IPAddressParsing))]
```

Comment:
Does this include an empty string?  If not, it'd be good to augment it, since Parse explicitly relies on empty string parsing failing.

Reply:
IPAddressParsing.InvalidIpv4AddressesStandalone does

https://github.com/dotnet/corefx/blob/07f9031bd334c8c461916cfa7d538c307aa484bd/src/System.Net.Primitives/tests/FunctionalTests/IPAddressParsing.cs#L129